File: js/class/GConfig.js
//===============================================
var GConfig = (function() {
    //===============================================
    var m_instance;
    //===============================================
    var Container = function() {
        return {
            //===============================================
            init: function() {
				var lEditorDir = localStorage.getItem("EditorDir");
				var lEditorFile = localStorage.getItem("EditorFile");     
				var lFormulaText = localStorage.getItem("FormulaText");     
				if(!lEditorDir || lEditorDir == "undefined") localStorage.setItem("EditorDir", "");
				if(!lEditorFile || lEditorFile == "undefined") localStorage.setItem("EditorFile", "");			
				if(!lFormulaText || lFormulaText == "undefined") localStorage.setItem("FormulaText", "");			
			},
            //===============================================
            setData: function(key, value) {
                localStorage.setItem(key, value);
            },
            //===============================================
            getData: function(key) {
                return localStorage.getItem(key);
            },
            //===============================================
        };
    }
    //===============================================
    return {
        Instance: function() {
            if(!m_instance) {
                m_instance = Container();
            }
            return m_instance;
        }
    };
    //===============================================
})();
//===============================================
GConfig.Instance().init();
//===============================================

File: Tutoriels/Cpp/Thread/code/src/manager/GConfig.cpp
//================================================
#include "GConfig.h"
//================================================
GConfig* GConfig::m_instance = 0;
//================================================
GConfig::GConfig() {
    setData("PROCESS_ID_SHOW_FLAG", "FALSE");
    setData("LIBRARY_LOAD_FLAG", "FALSE");
}
//================================================
GConfig::~GConfig() {

}
//================================================
GConfig* GConfig::Instance() {
    if(m_instance == 0) {
        m_instance = new GConfig;
    }
    return m_instance;
}
//================================================
void GConfig::setData(const QString& key, const QString& data) {
    m_data[key] = data;
}
//================================================
QString GConfig::getData(const QString& key) {
    return m_data[key];
}
//================================================
